R/delete_current_year_model_runs.R

```diff
@@ -0,0 +1,83 @@
+# Script to delete a list of model runs by ID from AWS.
+#
+# Accepts one argument, a comma-delimited list of run IDs for model runs
+# whose artifacts should be deleted.
+#
+# Assumes that model runs are restricted to the current assessment cycle, where
+# each assessment cycle starts in April. Raises an error if no objects matching
+# a given ID for the current year could be located in S3. This error will get
+# raised before any deletion occurs, so if one or more IDs are invalid then
+# no objects will be deleted.
+#
+# Example usage (delete the runs 123, 456, and 789 in the current year):
+#
+#   delete_current_year_model_runs.R 123,456,789
+
+library(glue)
+library(here)
+library(magrittr)
+source(here("R", "helpers.R"))
+
+current_date <- as.POSIXct(Sys.Date())
+current_month <- current_date %>% format("%m")
+current_year <- current_date %>% format("%Y")
+
+# The following heuristic determines the current upcoming assessment cycle year:
+#
+#   * From April to December (post assessment), `year` = next year
+#   * From January to March (during assessment), `year` = current year
```

Comment:
Does this heuristic seem reasonable to you?

Reply:
Totally! I would bump the cutoff month to May though, as sometimes we still do model touchup stuff in April.

Reply:
Done in a4895c20d2b44cad197df855ac7cdbeebbbe9616!